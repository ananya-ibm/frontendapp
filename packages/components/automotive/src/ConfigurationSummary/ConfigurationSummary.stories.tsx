/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { ConfigurationSummary } from './ConfigurationSummary';

export default {
  title: 'Components/Automotive/ConfigurationSummary',
  component: ConfigurationSummary
};

const storyProps = {
  image: '/static/automotive/exterior.png',
  summaryText:
    'The original DMC Delorean from the Back to the Future franchise, with the real ability to travel in time and Michael J Foxâ€™s phone number included.',
  configurationCode: 'waitaminutedoc',
  deliveryDate: 'June 2021',
  priceBreakdownText:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
  priceBreakdown: [
    {
      text: 'Base Price',
      amount: {
        prefix: '+',
        currency: 'GBP',
        value: 500
      }
    },
    {
      text: 'Exterior Additions',
      amount: {
        prefix: '+',
        currency: 'GBP',
        value: 300
      }
    },
    {
      text: 'Interior Additions',
      amount: {
        prefix: '+',
        currency: 'GBP',
        value: 50000
      }
    },
    {
      text: 'Other Stuff',
      amount: {
        prefix: '+',
        currency: 'GBP',
        value: 300
      }
    },
    {
      text: 'Total cost',
      amount: {
        prefix: '+',
        currency: 'GBP',
        value: 35500
      }
    }
  ],
  onSaveConfiguration: () => {},
  onDeleteConfiguration: () => {},
  summarySelections: [
    {
      title: 'Your chosen exterior',
      options: [
        {
          title: 'Colour',
          text: 'Blue',
          amount: {
            prefix: '+',
            currency: 'GBP',
            value: 300
          },
          thumbnail: '/static/automotive/thumbnails/Nicer-Blue.png',
          onProductSelectionChange: () => {},
          changeButtonText: 'Change selection'
        },
        {
          title: 'Wheels',
          text: 'Big',
          amount: {
            prefix: '+',
            currency: 'GBP',
            value: 300
          },
          thumbnail: '/static/automotive/thumbnails/Shiny-Black.png',
          onProductSelectionChange: () => {},
          changeButtonText: 'Change selection'
        }
      ]
    },
    {
      title: 'Your chosen interior',
      options: [
        {
          title: 'Seat',
          text: 'Leather',
          amount: {
            prefix: '+',
            currency: 'GBP',
            value: 300
          },
          thumbnail: '/static/automotive/thumbnails/Leather-Seat.png',
          onProductSelectionChange: () => {},
          changeButtonText: 'Change selection'
        },
        {
          title: 'Seat',
          text: 'Texture',
          amount: {
            prefix: '+',
            currency: 'GBP',
            value: 300
          },
          thumbnail: '/static/automotive/thumbnails/Waffle-Seat.png',
          onProductSelectionChange: () => {},
          changeButtonText: 'Change selection'
        },
        {
          title: 'Tyre',
          text: 'Round',
          version: 'R16 Hot Wheel',
          thumbnail: '/static/automotive/thumbnails/Forest-Green-Wheel.png',
          onProductSelectionChange: () => {},
          changeButtonText: 'Change selection'
        }
      ]
    },
    {
      title: 'Your chosen optionals',
      options: [
        {
          text: 'Winter Package',
          amount: {
            prefix: '+',
            currency: 'GBP',
            value: 300
          },
          thumbnail: '/static/automotive/thumbnails/Time-Travel-GPS.png',
          onProductSelectionChange: () => {},
          changeButtonText: 'Change selection'
        },
        {
          title: 'Colour',
          text: 'Yellow',
          amount: {
            prefix: '+',
            currency: 'GBP',
            value: 300
          },
          thumbnail: '/static/automotive/thumbnails/yellow.png',
          onProductSelectionChange: () => {},
          changeButtonText: 'Change selection'
        }
      ]
    }
  ]
};

export const standard = args => <ConfigurationSummary {...args} />;
standard.args = storyProps;

export const small = args => <ConfigurationSummary {...args} />;
small.args = {
  ...storyProps,
  isSmall: true
};
