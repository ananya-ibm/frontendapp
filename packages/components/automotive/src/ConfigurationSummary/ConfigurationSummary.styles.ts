/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import styled, { css } from 'styled-components';
import { ifProp } from '@exo/frontend-common-style-utils';
import theme from './ConfigurationSummary.theme';

export const ConfigurationSummary = styled('div')`
  display: flex;
  max-width: ${props => theme(props).maxWidth};
`;

export const LeftSummary = styled('div')`
  background-color: ${props => theme(props).leftBackground};
  flex: 0 0 50%;
  padding: ${props => theme(props).layoutSpacing};
`;
export const RightSummary = styled('div')`
  background-color: ${props => theme(props).rightBackground};
  flex: 0 0 50%;
  padding: ${props => theme(props).layoutSpacing};

  & .selection {
    margin: ${props => theme(props).paragraphSpacing} 0 0;
  }
`;

export const SummarySection = styled('div')`
  display: flex;
  flex: 0 50%;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: ${props => theme(props).paragraphSpacing} 0 0;
`;

export const SummaryTile = styled('div')`
  flex: 0 50%;
  padding: ${props => theme(props).paragraphSpacing} 0;
`;

export const Title = styled('h2')<{ isSmall?: boolean }>`
  font: 700 ${props => theme(props).titleFontSize} ${props => theme(props).titleFontFamily};
  ${props => ifProp(props, 'isSmall').isTruthy().then(css`
    margin-top: 0;
  `)}
  ${props => ifProp(props, 'isSmall').isFalsy().then(css`
    margin-top: ${theme(props).paragraphSpacing};
  `)}
`;

export const Text = styled('div')`
  font: 400 ${props => theme(props).textFontSize} ${props => theme(props).textFontFamily};
  margin-top: ${props => theme(props).paragraphSpacing};
`;

export const BoldText = styled('div')`
  font-weight: 700;
`;

export const Subtitle = styled('h4')<{ newLine?: boolean }>`
  display: ${props => (props.newLine ? 'block' : 'inline')};
  font: 700 ${props => theme(props).textFontSize} ${props => theme(props).textFontFamily};
  margin-top: ${props => theme(props).layoutSpacing};
`;

export const ButtonGroup = styled('div')`
  align-items: center;
  display: flex;
  justify-content: flex-end;
  margin: ${props => props.theme.spacing.stack.s7} 0 0;
`;

export const Price = styled('div')`
  margin: ${props => props.theme.spacing.stack.lM} 0 0;
`;

export const Button = styled('div')`
  margin-left: ${props => theme(props).layoutSpacing};
`;

export const Image = styled('img')`
  margin-top: ${props => theme(props).paragraphSpacing};
  width: 100%;
`;

export const SmallConfigurationSummary = styled('div')`
  background-color: ${props => theme(props).background};
  padding: ${props => theme(props).tilePadding};
`;
