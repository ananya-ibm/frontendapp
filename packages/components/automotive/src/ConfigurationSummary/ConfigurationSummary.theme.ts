/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

type Props = {
  background: string;
  leftBackground: string;
  rightBackground: string;
  titleFontSize: string;
  titleFontFamily: string;
  textFontSize: string;
  textFontFamily: string;
  layoutSpacing: string;
  paragraphSpacing: string;
  maxWidth: string;
  tilePadding: string;
};

declare global {
  interface EXOComponentStyles {
    automotive_configurationSummary?: Partial<Props>;
  }
}

export default (props: EXOThemeProps): Props => {
  return {
    // Default theme
    background: props.theme.colors.backgrounds.panels.secondary.base,
    leftBackground: props.theme.colors.backgrounds.panels.secondary.base,
    rightBackground: props.theme.colors.backgrounds.panels.secondary.base,
    titleFontSize: props.theme.typography.heading.heading3.size?.toString(),
    titleFontFamily: props.theme.typography.heading.heading3.family,
    textFontSize: '1rem', // props.theme.typography.body.short.size,
    textFontFamily: props.theme.typography.body.short.M.family,
    layoutSpacing: props.theme.spacing.stack.s7,
    paragraphSpacing: props.theme.spacing.stack.s5,
    maxWidth: '74rem',
    tilePadding: props.theme.spacing.inset.XL,

    // Override with custom theme - this also defines the key under which
    // the theme can be overridden, in this case .automotive.configurationSummary
    // in general should follow the format .<domain>.<componentName>
    ...props.theme?.byComponent?.automotive_configurationSummary
  };
};
