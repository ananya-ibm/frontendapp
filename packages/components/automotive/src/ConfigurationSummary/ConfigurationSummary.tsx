/* eslint-disable react/jsx-indent */
/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { MonetaryAmount, PriceTable } from '@exo/frontend-components-core';
import { Button } from '@exo/frontend-components-base';
import { addImageExt } from '@exo/frontend-common-utils';
import { SummaryTile } from '../SummaryTile/SummaryTile';
import * as S from './ConfigurationSummary.styles';

export const ConfigurationSummary = ({
  summaryText,
  configurationCode,
  deliveryDate,
  priceBreakdownText,
  priceBreakdown,
  onSaveConfiguration,
  onDeleteConfiguration,
  summarySelections,
  CTAbutton,
  image,
  isSmall,
  isSaving
}: Props) => {
  // eslint-disable-next-line react/jsx-indent
  return isSmall ? (
    <S.SmallConfigurationSummary>
      <S.Title isSmall>Your Car</S.Title>
      <S.Text>{summaryText}</S.Text>
      <S.Text>
        On the road price:
        <S.BoldText>
          <MonetaryAmount {...priceBreakdown[priceBreakdown.length - 1].amount} />
        </S.BoldText>
      </S.Text>
    </S.SmallConfigurationSummary>
  ) : (
    <S.ConfigurationSummary>
      <S.LeftSummary>
        <S.Title>Your new car</S.Title>
        {image && <S.Image src={addImageExt(image)} />}
        <S.Text>{summaryText}</S.Text>
        <S.Text>
          Your configuration code:
          <strong> {configurationCode}</strong>
        </S.Text>
        {deliveryDate && (
          <S.Text>
            Estimated delivery date:
            <strong> {deliveryDate}</strong>
          </S.Text>
        )}
        {priceBreakdown && (
          <S.Price>
            <S.Title>Price breakdown</S.Title>
            <S.Text>{priceBreakdownText}</S.Text>
            <PriceTable priceBreakdown={priceBreakdown} />
          </S.Price>
        )}
        <S.ButtonGroup>
          <>
            {CTAbutton ? (
              CTAbutton()
            ) : (
              <Button disabled={isSaving} onClick={onSaveConfiguration} label="Save Configuration" />
            )}
            {onDeleteConfiguration && (
              <S.Button>
                <Button variant="link" label="Delete Configuration" onClick={onDeleteConfiguration} />
              </S.Button>
            )}
          </>
        </S.ButtonGroup>
      </S.LeftSummary>
      {summarySelections && (
        <S.RightSummary>
          <S.Title>Summary</S.Title>
          {summarySelections.map(selection => (
            <div key={`sel-${selection.title}`} className="selection">
              <S.Subtitle>{selection.title}</S.Subtitle>
              <S.SummarySection>
                {selection.options.map(option => (
                  <S.SummaryTile key={`opt-${option.title}-${option.text}`}>
                    <SummaryTile
                      title={option.title}
                      text={option.text}
                      amount={option.amount}
                      version={option.version}
                      thumbnail={addImageExt(option.thumbnail)}
                      onChange={option.onProductSelectionChange}
                      changeButtonText={option.changeButtonText}
                    />
                  </S.SummaryTile>
                ))}
              </S.SummarySection>
            </div>
          ))}
        </S.RightSummary>
      )}
    </S.ConfigurationSummary>
  );
};

type Props = {
  isSmall?: boolean;
  isSaving?: boolean;
  CTAbutton?: () => void;
  image?: string;
  summaryText: string;
  configurationCode: string;
  deliveryDate: string;
  priceBreakdownText?: string;
  priceBreakdown: {
    text?: string;
    amount: {
      prefix?: string;
      currency?: string;
      value: string | number;
    };
  }[];
  onSaveConfiguration?: () => void;
  onDeleteConfiguration?: () => void;
  summarySelections: {
    title?: string;
    options: {
      title: string;
      text?: string;
      amount: {
        prefix?: string;
        currency?: string;
        value: number | string;
      };
      version?: string;
      thumbnail?: string;
      onProductSelectionChange: (arg: any) => void;
      changeButtonText: string;
    }[];
  }[];
};
