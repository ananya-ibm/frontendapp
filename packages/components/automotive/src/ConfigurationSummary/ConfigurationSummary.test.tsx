/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { renderWithTheme as render } from '@testUtils';
import { ConfigurationSummary } from './ConfigurationSummary';

const testProps = {
  summaryText:
    'The original DMC Delorean from the Back to the Future franchise, with the real ability to travel in time and Michael J Foxâ€™s phone number included.',
  configurationCode: 'waitaminutedoc',
  deliveryDate: 'June 2021',
  priceBreakdownText: 'Lorem ipsum dolor sit amet',
  priceBreakdown: [
    {
      text: 'Base Price',
      amount: {
        prefix: '+',
        currency: 'GBP',
        value: 500
      }
    }
  ],
  onSaveConfiguration: () => {},
  onDeleteConfiguration: () => {},
  summarySelections: [
    {
      title: 'Your chosen exterior',
      options: [
        {
          title: 'Colour 1',
          text: 'Blue',
          amount: {
            prefix: '+',
            currency: 'GBP',
            value: 300
          },
          thumbnail: '/yellow.png',
          onProductSelectionChange: () => {},
          changeButtonText: 'Change selection'
        },
        {
          title: 'Colour 2',
          text: 'Yellow',
          amount: {
            prefix: '+',
            currency: 'GBP',
            value: 300
          },
          thumbnail: '/yellow.png',
          onProductSelectionChange: () => {},
          changeButtonText: 'Change selection'
        }
      ]
    }
  ]
};

describe('<ConfigurationSummary /> component', () => {
  test('its snapshot matches', () => {
    const { container } = render(<ConfigurationSummary {...testProps} />);

    expect(container).toBeDefined();
    expect(container.firstChild).toMatchSnapshot();
  });
});
