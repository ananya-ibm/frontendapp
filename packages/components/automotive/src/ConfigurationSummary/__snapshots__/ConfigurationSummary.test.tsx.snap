// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfigurationSummary /> component its snapshot matches 1`] = `
.c12 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  color: #0f62fe;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font: bold 0.875rem/1.125005rem 'IBM Plex Sans',Arial,sans-serif;
  -webkit-letter-spacing: 0.16px;
  -moz-letter-spacing: 0.16px;
  -ms-letter-spacing: 0.16px;
  letter-spacing: 0.16px;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c12:hover {
  color: #0043ce;
}

.c12 svg {
  color: #161616;
}

.c10 {
  font-size: 105%;
  font-weight: 700;
  line-height: 0;
  white-space: nowrap;
}

.c10 svg {
  max-height: 100%;
}

.c8 .price-list {
  color: rgba(22,22,22,0.25);
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

.c5 {
  border-bottom: 0.0625rem solid #f4f4f4;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
}

.c6 {
  padding: 1rem 0;
}

.c7 {
  border-bottom: 0;
  font-weight: 700;
  padding: 1rem 0;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 6.25rem;
  width: 15rem;
}

.c18 {
  background: url(/yellow.png) no-repeat top center;
  background-size: contain;
  height: 0;
  max-width: 6rem;
  overflow: hidden;
  padding-top: 100%;
  width: 100%;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  padding: 0 1rem;
}

.c20 {
  font: 400 1rem/1.5rem 'IBM Plex Sans',Arial,sans-serif;
}

.c22 {
  font: 400 0.875rem/1.125005rem 'IBM Plex Sans',Arial,sans-serif;
  -webkit-letter-spacing: 0.16px;
  -moz-letter-spacing: 0.16px;
  -ms-letter-spacing: 0.16px;
  letter-spacing: 0.16px;
}

.c21 {
  font: 600 0.875rem/1.125005rem 'IBM Plex Sans',Arial,sans-serif;
  -webkit-letter-spacing: 0.16px;
  -moz-letter-spacing: 0.16px;
  -ms-letter-spacing: 0.16px;
  letter-spacing: 0.16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 74rem;
}

.c1 {
  background-color: #ffffff;
  -webkit-flex: 0 0 50%;
  -ms-flex: 0 0 50%;
  flex: 0 0 50%;
  padding: 2rem;
}

.c13 {
  background-color: #ffffff;
  -webkit-flex: 0 0 50%;
  -ms-flex: 0 0 50%;
  flex: 0 0 50%;
  padding: 2rem;
}

.c13 .selection {
  margin: 1rem 0 0;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 50%;
  -ms-flex: 0 50%;
  flex: 0 50%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 1rem 0 0;
}

.c16 {
  -webkit-flex: 0 50%;
  -ms-flex: 0 50%;
  flex: 0 50%;
  padding: 1rem 0;
}

.c2 {
  font: 700 1.75rem 'IBM Plex Sans',Arial,sans-serif;
  margin-top: 1rem;
}

.c3 {
  font: 400 1rem 'IBM Plex Sans',Arial,sans-serif;
  margin-top: 1rem;
}

.c14 {
  display: inline;
  font: 700 1rem 'IBM Plex Sans',Arial,sans-serif;
  margin-top: 2rem;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin: 2rem 0 0;
}

.c4 {
  margin: 3rem 0 0;
}

.c11 {
  margin-left: 2rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
    >
      Your new car
    </h2>
    <div
      class="c3"
    >
      The original DMC Delorean from the Back to the Future franchise, with the real ability to travel in time and Michael J Fox’s phone number included.
    </div>
    <div
      class="c3"
    >
      Your configuration code:
      <strong>
         
        waitaminutedoc
      </strong>
    </div>
    <div
      class="c3"
    >
      Estimated delivery date:
      <strong>
         
        June 2021
      </strong>
    </div>
    <div
      class="c4"
    >
      <h2
        class="c2"
      >
        Price breakdown
      </h2>
      <div
        class="c3"
      >
        Lorem ipsum dolor sit amet
      </div>
      <div
        class="PriceTablestyles__PriceTable-sc-12x6422-0"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            Base Price
          </div>
          <div
            class="c7"
          >
            <span
              class="c8"
            >
              <span
                class="prefix"
              >
                +
              </span>
              <span
                class="value"
              >
                £500
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c9"
    >
      <button
        class="c10 cds--btn cds--btn--primary"
        tabindex="0"
        type="button"
      >
        Save Configuration
         
      </button>
      <div
        class="c11"
      >
        <a
          class="c12"
        >
          Delete Configuration
        </a>
      </div>
    </div>
  </div>
  <div
    class="c13"
  >
    <h2
      class="c2"
    >
      Summary
    </h2>
    <div
      class="selection"
    >
      <h4
        class="c14"
      >
        Your chosen exterior
      </h4>
      <div
        class="c15"
      >
        <div
          class="c16"
        >
          <div
            class="c17"
          >
            <div
              class="c18"
            />
            <div
              class="c19"
            >
              <div
                class="c20"
              >
                 
                Colour 1
                 
              </div>
              <div>
                <div
                  class="c21"
                >
                  Blue
                   
                </div>
                <div
                  class="c22"
                >
                  <span
                    class="c8"
                  >
                    <span
                      class="prefix"
                    >
                      +
                    </span>
                    <span
                      class="value"
                    >
                      £300
                    </span>
                  </span>
                </div>
              </div>
              <a
                class="c12"
              >
                Change selection
              </a>
            </div>
          </div>
        </div>
        <div
          class="c16"
        >
          <div
            class="c17"
          >
            <div
              class="c18"
            />
            <div
              class="c19"
            >
              <div
                class="c20"
              >
                 
                Colour 2
                 
              </div>
              <div>
                <div
                  class="c21"
                >
                  Yellow
                   
                </div>
                <div
                  class="c22"
                >
                  <span
                    class="c8"
                  >
                    <span
                      class="prefix"
                    >
                      +
                    </span>
                    <span
                      class="value"
                    >
                      £300
                    </span>
                  </span>
                </div>
              </div>
              <a
                class="c12"
              >
                Change selection
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
